### DISCUSSION TITLE
Limit text in a TE script - only want X rows of text

### DISCUSSION THREAD

**Discussion by edhans** [TabularEditor/TabularEditor3#460] [🙋♀️TE3 Q&A] [answered]
Hi all, question.

I use this script in Tabular Editor to format my DAX code and then add the code to the description field so that when you hover over it in Power BI desktop in both the field pane or the field well of a visual, you can see the code. 
```
`FormatDax ( Model.AllMeasures ,  true);
foreach  (var  m  in  Model.AllMeasures )
     {
         m.Expression   =   '\n'   +   m.Expression ;
         m.Description   =   m.Expression ;
     }
```
The problem is if it is a really large measure with many variable declarations and well formatted code, it can run to 50+ lines easily. Often when that happens what you see in the hover-over tooltip is truncated at both the top and bottom.

I'd like to limit my m.Description to preferably a row limit, perhaps 20, and if not that, a character limit, though that would be less useful. Rows would be ideal, but I suspect you'd have to get into counting line feeds in the expression.
`

**Comment by otykier** [comment]
By the power of LINQ in C# this can be achieved with:

```csharp
FormatDax (Model.AllMeasures, true);
foreach (var m in Model.AllMeasures)
{
    m.Expression = '\n' + m.Expression;
    m.Description = string.Join('\n', m.Expression.Split('\n').Take(20).ToArray());
}
```

**Reply by edhans** [reply]
Hmmm... won't compile.
<img width="785" alt="image" src="https://user-images.githubusercontent.com/45298358/157749918-21bc8d91-a244-4297-82fa-660126291a3d.png">
Seems like it is saying it cannot convert '\n' to a string. That makes no sense to me, but I know zilch about LINQ and C#

**Comment by otykier** [comment, marked as answer]
Sorry, my mistake. Use `"\n"` instead of `'\n'`.
